.github/workflows/build.yaml

```diff
@@ -69,10 +69,17 @@ jobs:
 
       - uses: actions/upload-artifact@v4
         with:
-          name: jvm-native-libraries
+          name: jvm-native-libraries-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.cmake-arch }}
```

Comment:
This should be reverted. We want them to merge together.

Reply:
it's [failing](https://github.com/cashapp/zipline/actions/runs/7228285030/job/19697454944#step:9:15) with the [same reason](https://github.com/cashapp/zipline/pull/1200#discussion_r1428571699) after revert, I found [this](https://github.com/actions/upload-artifact?tab=readme-ov-file#not-uploading-to-the-same-artifact) from the `upload-artifact` documentation about `V4`:

<img width="865" alt="Screenshot 2023-12-15 at 4 44 35 PM" src="https://github.com/cashapp/zipline/assets/57155915/b288ffc0-2005-43ff-8d5b-7797da00be78">

To keep the merge together behavior, I'll revert the job uploading `jvm-native-libraries` to use `upload-artifact` V3.

Reply:
We can still change them to be unique, but it means the consuming job will need to download them all and merge locally.

Reply:
I'll keep it to V3 for this PR to get the uploading test files part in, it sounds moving forward: 

1) this should be updated to V4 with unique names 
2) update the consuming job to download and merge (where do we have the consuming jobs right now?)

Reply:
Consuming job is probably named publish. Whichever has the download action for this folder.